### PR TITLE
Merging danielBranch to master. 

### PR DESCRIPTION
**Pull request by LHuss**
This branch focused mainly on issue #9.
Since the last pull request from this branch, a small search functionality was added to the listings page.

### CODE REVIEW

File: app/src/main/java/Fragments/HomeFragment.java [hunk 1]
```diff
@@ -86,4 +107,80 @@ public void onCancelled(DatabaseError databaseError) {
...
+						if (!isViewFiltered || isViewFiltered && item.getName().toLowerCase().contains(filterString)) {
+							listingsList.add(item);
+						}
+					}
+					mAdapter.notifyDataSetChanged();
+					return false;
+				}
+
+			    @Override
+			    public boolean onQueryTextChange(String newText) {
```
Comment (EvanMateo7) [inline app/src/main/java/Fragments/HomeFragment.java:142]: // Reset listing list on empty search

**Review by EvanMateo7 — APPROVED** [review]
Will need to add more comments to describe functions and purposes of some code blocks.

File: app/src/main/java/Fragments/HomeFragment.java [hunk 2]
```diff
@@ -22,11 +27,22 @@
...
 	private ArrayList<Listing> listingsList = new ArrayList<>();
 	private RecyclerView recyclerView;
 	private ListItemAdapter mAdapter;
+	private boolean isViewFiltered;
+	private String filterString;
+
+	// Fragment View
+	private Context context;
+
+	MenuItem fav;
```
Comment (kkhh001) [inline app/src/main/java/Fragments/HomeFragment.java]: Is the variable fav for the favourite feature which will be implemented later?

**Review by kkhh001 — APPROVED** [review]
I agree with Evan. Other than my question in the code, it looks good!

File: app/src/main/java/Fragments/HomeFragment.java (hunk 2 above)
Comment (LHuss) [reply app/src/main/java/Fragments/HomeFragment.java]: Unsure. @danielanthonyr ^?

**Review by danielanthonyr — COMMENTED** [review]
Good stuff. Clean code

**Pull request merged**